sale_automatic_workflow_job/models/automatic_workflow_job.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2020 Camptocamp (https://www.camptocamp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import models
+
+from odoo.addons.queue_job.job import identity_exact
+
+
+class AutomaticWorkflowJob(models.Model):
+    _inherit = "automatic.workflow.job"
+
+    def _do_validate_sale_order_job_options(self, sale, domain_filter):
+        description = self.env._("Validate sales order %s", sale.display_name)
+        return {
+            "description": description,
+            "identity_key": identity_exact,
+        }
+
+    def _validate_sale_orders(self, domain_filter):
```

Comment:
It breaks the flows of sending the order confirmation email. Look at the original method:
```
    @api.model
    def _validate_sale_orders(self, order_filter):
        sale_obj = self.env["sale.order"]
        sales = sale_obj.search(order_filter)
        _logger.debug("Sale Orders to validate: %s", sales.ids)
        for sale in sales:
            with savepoint(self.env.cr):
                self._do_validate_sale_order(
                    sale.with_company(sale.company_id), order_filter
                )
                if self.env.context.get("send_order_confirmation_mail"):
                    self._do_send_order_confirmation_mail(sale)
```
When `def_do_validate_sale_order()` does not change order's state to "sale" right away. 
So `def _do_send_order_confirmation_mail()` does not send email anymore

Reply:
I add `_do_send_order_confirmation_mail` in job and it will be runned after 2 minutes